From 17e051cf3ea2421bb094d00d85d71761349e9833 Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Thu, 13 Oct 2016 11:53:58 +0100
Subject: [PATCH] build: Pass 'CC' to the introspection scanner

This avoids g-ir-scanner picking up a different compiler from the
environment than the one specified when building.

The behaviour is the same as the one provided by GObject introspection
itself in its autotools integration, except it's missing CFLAGS,
CPPFLAGS, and LDFLAGS, because their presence makes the command line too
long.
---
 cmake/modules/GObjectIntrospectionMacros.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/GObjectIntrospectionMacros.cmake b/cmake/modules/GObjectIntrospectionMacros.cmake
index 98a38e4..33e2c0f 100644
--- a/cmake/modules/GObjectIntrospectionMacros.cmake
+++ b/cmake/modules/GObjectIntrospectionMacros.cmake
@@ -50,7 +50,8 @@ macro(gir_add_introspections introspections_girs)
     set(_gir_libtool "--no-libtool")
 
     add_custom_command(
-      COMMAND ${INTROSPECTION_SCANNER}
+      COMMAND ${CMAKE_COMMAND} -E env "CC='${CMAKE_C_COMPILER}'"
+              ${INTROSPECTION_SCANNER}
               ${INTROSPECTION_SCANNER_ARGS}
               --namespace=${_gir_namespace}
               --nsversion=${_gir_version}
-- 
2.9.3

